art/estimators/certification/derandomized_smoothing/derandomized_smoothing_tensorflow.py

```diff
@@ -24,119 +24,11 @@
-        else:
-            raise ValueError("Ablation type not supported. Must be either column or block")
-
-    def _predict_classifier(self, x: np.ndarray, batch_size: int, training_mode: bool, **kwargs) -> np.ndarray:
-        """
-        Perform prediction for a batch of inputs.
-
-        :param x: Input samples.
-        :param batch_size: Size of batches.
-        :param training_mode: `True` for model set to training mode and `'False` for model set to evaluation mode.
-        :return: Array of predictions of shape `(nb_inputs, nb_classes)`.
-        """
-        raise NotImplementedError
-
-    def predict(self, x: np.ndarray, batch_size: int = 128, training_mode: bool = False, **kwargs) -> np.ndarray:
-        """
-        Performs cumulative predictions over every ablation location
-
-        :param x: Unablated image
-        :param batch_size: the batch size for the prediction
-        :param training_mode: if to run the classifier in training mode
-        :return: cumulative predictions after sweeping over all the ablation configurations.
-        """
-        if self._channels_first:
-            columns_in_data = x.shape[-1]
-            rows_in_data = x.shape[-2]
-        else:
-            columns_in_data = x.shape[-2]
-            rows_in_data = x.shape[-3]
-
-        if self.ablation_type in {"column", "row"}:
-            if self.ablation_type == "column":
-                ablate_over_range = columns_in_data
-            else:
-                # image will be transposed, so loop over the number of rows
-                ablate_over_range = rows_in_data
-
-            for ablation_start in range(ablate_over_range):
-                ablated_x = self.ablator.forward(np.copy(x), column_pos=ablation_start)
-                if ablation_start == 0:
-                    preds = self._predict_classifier(
-                        ablated_x, batch_size=batch_size, training_mode=training_mode, **kwargs
-                    )
-                else:
-                    preds += self._predict_classifier(
-                        ablated_x, batch_size=batch_size, training_mode=training_mode, **kwargs
-                    )
-        elif self.ablation_type == "block":
-            for xcorner in range(rows_in_data):
-                for ycorner in range(columns_in_data):
-                    ablated_x = self.ablator.forward(np.copy(x), row_pos=xcorner, column_pos=ycorner)
-                    if ycorner == 0 and xcorner == 0:
-                        preds = self._predict_classifier(
-                            ablated_x, batch_size=batch_size, training_mode=training_mode, **kwargs
-                        )
-                    else:
-                        preds += self._predict_classifier(
-                            ablated_x, batch_size=batch_size, training_mode=training_mode, **kwargs
-                        )
-        return preds
+import tensorflow as tf
```

Comment:
Check that it does not introduce a required dependency.

Reply:
updated to avoid top level import.